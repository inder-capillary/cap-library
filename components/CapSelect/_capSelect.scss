@import "../styles/_variables.scss";

$select: cap-select-v2;

.ant-select.#{$select} {
  .ant-select-arrow {
    right: 4px;
    margin-top: 0;
    transform: translateY(-50%);
    .ant-select-arrow-icon {
      line-height: 16px;
    }
  }

  &.ant-select-enabled .ant-select-arrow {
    color: $CAP_G01;
  }

  &.ant-select-open .ant-select-arrow-icon svg {
    transform: none;
  }

  .ant-select-selection--single {
    height: 40px;
  }

  .ant-select-selection__rendered {
    line-height: 40px;
  }

  &.ant-select-enabled .ant-select-selection {
    border: 1px solid $CAP_G06;
    &:hover {
      border: 1px solid $CAP_G11;
    }
  }

  &.ant-select-enabled .ant-select-focused .ant-select-selection,
  &.ant-select-enabled .ant-select-selection:focus,
  &.ant-select-enabled .ant-select-selection:active {
    border-color: $CAP_G01;
    box-shadow: none;
  }

  &.ant-select-disabled {
    border-color: $CAP_G07;
    .ant-select-selection {
      background-color: transparent;
    }
  }
}
.ant-select-dropdown.#{$select}-dropdown {
  .ant-select-dropdown-menu-item {
    padding: 8px 24px;
    color: $FONT_COLOR_02;
    &:hover {
      background-color: $BG_01;
    }
    &.ant-select-dropdown-menu-item-selected {
      background-color: map-get($CAP_SECONDARY, light);
      color: map-get($CAP_SECONDARY, base);
      font-weight: $FONT_WEIGHT_MEDIUM;
      &::before {
        content: " ";
        width: 2px;
        height: 100%;
        left: 0;
        top: 0;
        position: absolute;
        background-color: map-get($CAP_SECONDARY, base);
      }
    }
  }
}
