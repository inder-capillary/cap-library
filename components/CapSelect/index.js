/**
*
* CapSelect
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Select } from 'antd';
import CapCustomSelect from '../CapCustomSelect';
import CapIcon from '../CapIcon';
import './_capSelect.scss';
import isEmpty from 'lodash/isEmpty';


const clsPrefix = 'cap-select-v2';
const { Option } = Select;

function CapSelect(props) {
  const { className, dropdownClassName, options, ...rest } = props;
  const items = options.map((op) => <Option {...op}>{op.label}</Option>);
  return (
    <div>
      {isEmpty(props.label) && <h4>{props.label}</h4>}
      <Select
        removeIcon={<CapIcon type="close" size="s" />}
        suffixIcon={<CapIcon type="chevron-down" />}
        menuItemSelectedIcon={<CapIcon type="tick" />}
        dropdownClassName={classNames(`${clsPrefix}-dropdown`, dropdownClassName)}
        {...rest}
        className={classNames(clsPrefix, className)}>
        {items}
      </Select>
      {isEmpty(props.errorMessage) && <span className="capSelect-error-message">{props.errorMessage}</span>}
    </div>
  );
}

CapSelect.defaultProps = {
  size: 'large',
};

CapSelect.propTypes = {
  className: PropTypes.string,
  options: PropTypes.array.isRequired,
  size: PropTypes.string,
};

CapSelect.CapCustomSelect = CapCustomSelect;

export default CapSelect;
