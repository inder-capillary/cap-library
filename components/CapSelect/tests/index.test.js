import React from 'react';
import { shallow } from 'enzyme';

import CapSelect from '../index';

const orgsList = [
  { label: 'Purples', value: 'purples' },
  { label: 'Buckle', value: 'buckle' },
  { label: 'Ski', value: 'ski' },
  { label: 'Splash', value: 'splash' },
  { label: 'Metro', value: 'metro' },
];

describe('<CapSelect />', () => {
  const handleOrgChange = jest.fn();

  const defaultProps = {
    width: "250px",
    selectPlaceholder: "Select organizations",
    showSearch: true,
    options: orgsList,
    value: "Ski",
    onChange: handleOrgChange,
  };
  const wrapper = shallow(<CapSelect {...defaultProps} />);

  test('render capSelect', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should update search state when handleSearchFunction is called', () => {
    const event = { target: { value: "ski" } };
    wrapper.instance().handleSearch(event);
    expect(wrapper.state('searchText')).toEqual('ski');
  });

  it('Should Change popover visibility', () => {
    wrapper.setState({ visible: false });
    wrapper.instance().onVisibleChange(true);
    expect(wrapper.state('visible')).toEqual(true);
  });

  it('Should empty searchText if popover is closed', () => {
    wrapper.instance().onVisibleChange(false);
    expect(wrapper.state('visible')).toEqual(false);
    expect(wrapper.state('searchText')).toEqual("");
  });

  it('Should call handleOrgChange prop and close popover when an option is selected', () => {
    wrapper.setState({ visible: true });
    wrapper.instance().handleChange({ label: 'Purples', value: 'purples' });
    expect(wrapper.state('visible')).toEqual(false);
    expect(handleOrgChange).toHaveBeenCalled();
  });
});
