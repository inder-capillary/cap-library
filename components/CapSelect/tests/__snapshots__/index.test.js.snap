// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapSelect /> render capSelect 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CapSelect
    handleItemChange={[MockFunction]}
    items={
      Array [
        Object {
          "label": "Purples",
          "value": "purples",
        },
        Object {
          "label": "Buckle",
          "value": "buckle",
        },
        Object {
          "label": "Ski",
          "value": "ski",
        },
        Object {
          "label": "Splash",
          "value": "splash",
        },
        Object {
          "label": "Metro",
          "value": "metro",
        },
      ]
    }
    searchPlaceholder="Search"
    selectPlaceholder="Select organizations"
    selectedItem="Ski"
    showSearch={true}
    width="250px"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="cap-select-v2-selection"
        style={
          Object {
            "width": "250px",
          }
        }
      >
        <div
          className="cap-select-v2-selected-value"
        >
          <ForwardRef
            type="h5"
          >
            Select organizations
          </ForwardRef>
        </div>
        <img
          alt="down-icon"
          className="cap-select-v2-arrow"
          src="chevron-down.svg"
        />
      </div>,
      "content": <React.Fragment>
        <div
          className="cap-select-v2-search"
        >
          <Component
            onChange={[Function]}
            placeholder="Search"
            value=""
          />
          <img
            alt=""
            className="icon-search"
            src="search.svg"
          />
          
        </div>
        <div
          className="cap-select-v2-items-wrapper"
        >
          <div
            className="cap-select-v2-select-item"
            onClick={[Function]}
          >
            Purples
          </div>
          <div
            className="cap-select-v2-select-item"
            onClick={[Function]}
          >
            Buckle
          </div>
          <div
            className="cap-select-v2-select-item"
            onClick={[Function]}
          >
            Ski
          </div>
          <div
            className="cap-select-v2-select-item"
            onClick={[Function]}
          >
            Splash
          </div>
          <div
            className="cap-select-v2-select-item"
            onClick={[Function]}
          >
            Metro
          </div>
        </div>
      </React.Fragment>,
      "getPopupContainer": [Function],
      "mouseEnterDelay": 0.1,
      "mouseLeaveDelay": 0.1,
      "onVisibleChange": [Function],
      "overlayClassName": "cap-select-v2-popover",
      "overlayStyle": Object {
        "width": "100%",
      },
      "placement": "bottomLeft",
      "transitionName": "zoom-big",
      "trigger": "click",
      "visible": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cap-select-v2-selected-value"
          >
            <ForwardRef
              type="h5"
            >
              Select organizations
            </ForwardRef>
          </div>,
          <img
            alt="down-icon"
            className="cap-select-v2-arrow"
            src="chevron-down.svg"
          />,
        ],
        "className": "cap-select-v2-selection",
        "style": Object {
          "width": "250px",
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ForwardRef
              type="h5"
            >
              Select organizations
            </ForwardRef>,
            "className": "cap-select-v2-selected-value",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Select organizations",
              "type": "h5",
            },
            "ref": null,
            "rendered": "Select organizations",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "CapHeadingSpan": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "rules": Array [
                    "
font-size: ",
                    [Function],
                    ";
font-weight: ",
                    [Function],
                    ";
color: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "rules": Array [
                  "
    font-size: ",
                  [Function],
                  ";
    font-weight: ",
                  [Function],
                  ";
    color: ",
                  [Function],
                  ";
    ",
                ],
              },
              "defaultProps": Object {
                "type": "h5",
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "propTypes": Object {
                "children": [Function],
                "type": [Function],
              },
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "down-icon",
            "className": "cap-select-v2-arrow",
            "src": "chevron-down.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="cap-select-v2-selection"
          style={
            Object {
              "width": "250px",
            }
          }
        >
          <div
            className="cap-select-v2-selected-value"
          >
            <ForwardRef
              type="h5"
            >
              Select organizations
            </ForwardRef>
          </div>
          <img
            alt="down-icon"
            className="cap-select-v2-arrow"
            src="chevron-down.svg"
          />
        </div>,
        "content": <React.Fragment>
          <div
            className="cap-select-v2-search"
          >
            <Component
              onChange={[Function]}
              placeholder="Search"
              value=""
            />
            <img
              alt=""
              className="icon-search"
              src="search.svg"
            />
            
          </div>
          <div
            className="cap-select-v2-items-wrapper"
          >
            <div
              className="cap-select-v2-select-item"
              onClick={[Function]}
            >
              Purples
            </div>
            <div
              className="cap-select-v2-select-item"
              onClick={[Function]}
            >
              Buckle
            </div>
            <div
              className="cap-select-v2-select-item"
              onClick={[Function]}
            >
              Ski
            </div>
            <div
              className="cap-select-v2-select-item"
              onClick={[Function]}
            >
              Splash
            </div>
            <div
              className="cap-select-v2-select-item"
              onClick={[Function]}
            >
              Metro
            </div>
          </div>
        </React.Fragment>,
        "getPopupContainer": [Function],
        "mouseEnterDelay": 0.1,
        "mouseLeaveDelay": 0.1,
        "onVisibleChange": [Function],
        "overlayClassName": "cap-select-v2-popover",
        "overlayStyle": Object {
          "width": "100%",
        },
        "placement": "bottomLeft",
        "transitionName": "zoom-big",
        "trigger": "click",
        "visible": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cap-select-v2-selected-value"
            >
              <ForwardRef
                type="h5"
              >
                Select organizations
              </ForwardRef>
            </div>,
            <img
              alt="down-icon"
              className="cap-select-v2-arrow"
              src="chevron-down.svg"
            />,
          ],
          "className": "cap-select-v2-selection",
          "style": Object {
            "width": "250px",
          },
        },
        "ref": [Function],
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ForwardRef
                type="h5"
              >
                Select organizations
              </ForwardRef>,
              "className": "cap-select-v2-selected-value",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Select organizations",
                "type": "h5",
              },
              "ref": null,
              "rendered": "Select organizations",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "CapHeadingSpan": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "rules": Array [
                      "
font-size: ",
                      [Function],
                      ";
font-weight: ",
                      [Function],
                      ";
color: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "rules": Array [
                    "
    font-size: ",
                    [Function],
                    ";
    font-weight: ",
                    [Function],
                    ";
    color: ",
                    [Function],
                    ";
    ",
                  ],
                },
                "defaultProps": Object {
                  "type": "h5",
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "propTypes": Object {
                  "children": [Function],
                  "type": [Function],
                },
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "down-icon",
              "className": "cap-select-v2-arrow",
              "src": "chevron-down.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
